### PR TITLE
Fix a bug mentioned in the previous PR

### PR DESCRIPTION
**Pull request by ice1000**
Previous PR means #358

### CODE REVIEW

**Comment by ice1000** [comment]
[intellij-haskell.zip](https://github.com/rikvdkleij/intellij-haskell/files/2688638/intellij-haskell.zip)

**Comment by ice1000** [comment]
Seems it's not fixed.
DO NOT MERGE.

Notes:
https://github.com/JetBrains/intellij-community/blob/master/json/src/com/intellij/json/psi/JsonStringLiteralManipulator.java

**Comment by ice1000** [comment]
It's now working! It's very nice.

**Comment by rikvdkleij** [comment]
Cool! Thanks!

File: src/main/scala/intellij/haskell/psi/impl/HaskellStringLiteralManipulator.scala [hunk 1]
```diff
@@ -15,9 +15,13 @@ class HaskellStringLiteralManipulator extends AbstractElementManipulator[Haskell
...
                                    range: TextRange,
                                    newContent: String): HaskellStringLiteralElementImpl = {
+    val oldText = psi.getText
+    val newText = oldText.substring(0, range.getStartOffset) + newContent + oldText.substring(range.getEndOffset)
     val newElement = PsiFileFactory
       .getInstance(psi.getProject)
-      .createFileFromText("a.hs", HaskellLanguage.Instance, newContent, false, false)
+      .createFileFromText("a.hs", HaskellLanguage.Instance, newText, false, false)
+      .getLastChild
+      .getLastChild
```
[the remark this answers is not in the document]
Comment (rikvdkleij) [reply src/main/scala/intellij/haskell/psi/impl/HaskellStringLiteralManipulator.scala]: I also create elements in that way. See `HaskellElementFactory`. 

Can also be done by `createElementFromText` but that does not always work.

[the remark this answers is not in the document]
Comment (ice1000) [reply src/main/scala/intellij/haskell/psi/impl/HaskellStringLiteralManipulator.scala]: > I also create elements in that way. See HaskellElementFactory.

Maybe this code should be refactored to use that API.

**Pull request merged**